File: support/src/main/java/io/wax911/support/base/attribute/TimeTargetType.java
package io.wax911.support.base.attribute;

import androidx.annotation.IntDef;
import kotlin.annotation.AnnotationRetention;
import kotlin.annotation.Retention;

@Retention(AnnotationRetention.SOURCE)
@IntDef({TimeTargetType.TIME_UNIT_DAYS, TimeTargetType.TIME_UNIT_HOURS,
        TimeTargetType.TIME_UNIT_MINUTES, TimeTargetType.TIME_UNITS_SECONDS})
public @interface TimeTargetType {

    int TIME_UNIT_DAYS = 0,
            TIME_UNIT_HOURS = 1,
            TIME_UNIT_MINUTES = 2,
            TIME_UNITS_SECONDS = 3;
}

File: support/src/main/java/io/wax911/support/base/attribute/TimeTargetType.kt
package io.wax911.support.base.attribute

import androidx.annotation.IntDef

@Retention(AnnotationRetention.SOURCE)
@IntDef(TimeTargetType.TIME_UNIT_DAYS, TimeTargetType.TIME_UNIT_HOURS,
        TimeTargetType.TIME_UNIT_MINUTES, TimeTargetType.TIME_UNITS_SECONDS)
annotation class TimeTargetType {
    companion object {

        const val TIME_UNIT_DAYS = 0
        const val TIME_UNIT_HOURS = 1
        const val TIME_UNIT_MINUTES = 2
        const val TIME_UNITS_SECONDS = 3
    }
}
